NPE Crash after fresh installation due to #2680

Simply uninstall the client and reinstall it (from master) and it crashes right away:

```
java.lang.RuntimeException: Unable to instantiate activity ComponentInfo{com.nextcloud.client/com.owncloud.android.authentication.AuthenticatorActivity}: java.lang.NullPointerException: Attempt to invoke virtual method 'android.content.res.Resources android.content.Context.getResources()' on a null object reference
        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2679)
        at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2856)
        at android.app.ActivityThread.-wrap11(Unknown Source:0)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1589)
        at android.os.Handler.dispatchMessage(Handler.java:106)
        at android.os.Looper.loop(Looper.java:164)
        at android.app.ActivityThread.main(ActivityThread.java:6494)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:440)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:807)
     Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'android.content.res.Resources android.content.Context.getResources()' on a null object reference
        at android.content.ContextWrapper.getResources(ContextWrapper.java:89)
        at android.view.ContextThemeWrapper.getResourcesInternal(ContextThemeWrapper.java:127)
        at android.view.ContextThemeWrapper.getResources(ContextThemeWrapper.java:121)
        at android.support.v7.app.AppCompatActivity.getResources(AppCompatActivity.java:542)
        at com.owncloud.android.MainApp.getAccountType(MainApp.java:386)
        at com.owncloud.android.authentication.AuthenticatorActivity.<init>(AuthenticatorActivity.java:245)
        at java.lang.Class.newInstance(Native Method)
        at android.app.Instrumentation.newActivity(Instrumentation.java:1174)
        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2669)
        at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2856) 
        at android.app.ActivityThread.-wrap11(Unknown Source:0) 
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1589) 
        at android.os.Handler.dispatchMessage(Handler.java:106) 
        at android.os.Looper.loop(Looper.java:164) 
        at android.app.ActivityThread.main(ActivityThread.java:6494) 
        at java.lang.reflect.Method.invoke(Native Method) 
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:440) 
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:807) 
```

Crashing, changes line is:
https://github.com/nextcloud/android/pull/2680/files#diff-587dc61f19cacc80fdba456dee2cfeeaR245

introduced via #2680

--- a/build.gradle
+++ b/build.gradle
@@ -14,7 +14,7 @@ buildscript {
         google()
     }
     dependencies {
-        classpath 'com.android.tools.build:gradle:3.1.2'
+        classpath 'com.android.tools.build:gradle:3.1.3'
         classpath 'com.google.gms:google-services:3.0.0'
         classpath 'org.codehaus.groovy:groovy-all:2.4.12' 
         classpath('com.dicedmelon.gradle:jacoco-android:0.1.2') { 

--- a/src/main/java/com/owncloud/android/authentication/AuthenticatorActivity.java
+++ b/src/main/java/com/owncloud/android/authentication/AuthenticatorActivity.java
@@ -242,10 +242,9 @@ public class AuthenticatorActivity extends AccountAuthenticatorActivity
     /// Identifier of operation in progress which result shouldn't be lost 
     private long mWaitingForOpId = Long.MAX_VALUE;
 
-    private final String BASIC_TOKEN_TYPE = AccountTypeUtils.getAuthTokenTypePass(MainApp.getAccountType(this));
-    private final String OAUTH_TOKEN_TYPE = AccountTypeUtils.getAuthTokenTypeAccessToken(MainApp.getAccountType(this));
-    private final String SAML_TOKEN_TYPE = AccountTypeUtils.getAuthTokenTypeSamlSessionCookie(
-            MainApp.getAccountType(this));
+    private String basicTokenType;
+    private String oauthTokenType;
+    private String samlTokenType;
 
     private boolean webViewLoginMethod;
     private String webViewUser;
@@ -263,6 +262,10 @@ public class AuthenticatorActivity extends AccountAuthenticatorActivity
     protected void onCreate(Bundle savedInstanceState) {
         //Log_OC.e(TAG,  "onCreate init");
         super.onCreate(savedInstanceState);
+
+        basicTokenType = AccountTypeUtils.getAuthTokenTypePass(MainApp.getAccountType(this));
+        oauthTokenType = AccountTypeUtils.getAuthTokenTypeAccessToken(MainApp.getAccountType(this));
+        samlTokenType = AccountTypeUtils.getAuthTokenTypeSamlSessionCookie(MainApp.getAccountType(this));
 
         // delete cookies for webView
         deleteCookies();
@@ -561,11 +564,11 @@ public class AuthenticatorActivity extends AccountAuthenticatorActivity
 
     private String chooseAuthTokenType(boolean oauth, boolean saml) {
         if (saml) {
-            return SAML_TOKEN_TYPE;
+            return samlTokenType;
         } else if (oauth) {
-            return OAUTH_TOKEN_TYPE;
+            return oauthTokenType;
         } else {
-            return BASIC_TOKEN_TYPE;
+            return basicTokenType;
         }
     }
 
@@ -911,10 +914,10 @@ public class AuthenticatorActivity extends AccountAuthenticatorActivity
             String password = savedInstanceState.getString(KEY_PASSWORD);
 
             OwnCloudCredentials credentials = null;
-            if (BASIC_TOKEN_TYPE.equals(mAuthTokenType)) {
+            if (basicTokenType.equals(mAuthTokenType)) {
                 credentials = OwnCloudCredentialsFactory.newBasicCredentials(username, password);
 
-            } else if (OAUTH_TOKEN_TYPE.equals(mAuthTokenType)) {
+            } else if (oauthTokenType.equals(mAuthTokenType)) {
                 credentials = OwnCloudCredentialsFactory.newBearerCredentials(mAuthToken);
 
             }
@@ -1475,11 +1478,11 @@ public class AuthenticatorActivity extends AccountAuthenticatorActivity
 
 
     private boolean authSupported(AuthenticationMethod authMethod) {
-        return ((BASIC_TOKEN_TYPE.equals(mAuthTokenType) &&
+        return ((basicTokenType.equals(mAuthTokenType) &&
                 AuthenticationMethod.BASIC_HTTP_AUTH.equals(authMethod)) ||
-                (OAUTH_TOKEN_TYPE.equals(mAuthTokenType) &&
+                (oauthTokenType.equals(mAuthTokenType) &&
                         AuthenticationMethod.BEARER_TOKEN.equals(authMethod)) ||
-                (SAML_TOKEN_TYPE.equals(mAuthTokenType) &&
+                (samlTokenType.equals(mAuthTokenType) &&
                         AuthenticationMethod.SAML_WEB_SSO.equals(authMethod))
         );
     }
@@ -2020,9 +2023,9 @@ public class AuthenticatorActivity extends AccountAuthenticatorActivity
     public void onCheckClick(View view) {
         CheckBox oAuth2Check = (CheckBox) view;
         if (oAuth2Check.isChecked()) {
-            mAuthTokenType = OAUTH_TOKEN_TYPE;
+            mAuthTokenType = oauthTokenType;
         } else {
-            mAuthTokenType = BASIC_TOKEN_TYPE;
+            mAuthTokenType = basicTokenType;
         }
         updateAuthenticationPreFragmentVisibility();
     }

